src/pages/[platform]/build-a-backend/functions/examples/create_user_profile_record/index.mdx

```diff
@@ -0,0 +1,161 @@
+
+</Callout>
+
+To get started, create a new directory and a resource file, `amplify/auth/post-confirmation/resource.ts`. Then, define the Function with `defineFunction`:
+
+```ts title="amplify/auth/post-confirmation/resource.ts"
+import { defineFunction } from '@aws-amplify/backend';
+
+export const postConfirmation = defineFunction({
+  entry: './post-confirmation-handler.ts',
+});
+```
+
+Then, create the corresponding handler file, `amplify/auth/post-confirmation/post-confirmation-handler.ts`, file with the following contents:
+
+```ts title="amplify/auth/post-confirmation/handler.ts"
+import type { PostConfirmationTriggerHandler } from "aws-lambda";
+import { Amplify } from "aws-amplify";
+import { generateClient } from "aws-amplify/data";
+import { type Schema } from "../../data/resource";
+import { env } from "$amplify/env/post-confirmation-handler";
+import { modelIntrospection } from "./amplifyconfiguration.json";
+
+Amplify.configure(
+  {
+    API: {
+      GraphQL: {
+        endpoint: env.AMPLIFY_DATA_GRAPHQL_ENDPOINT, 
+        region: env.AWS_REGION,
+        defaultAuthMode: "iam",
+        modelIntrospection: modelIntrospection as never,
+      },
+    },
+  },
+  {
+    Auth: {
+      credentialsProvider: {
+        getCredentialsAndIdentityId: async () => ({
+          credentials: {
+            accessKeyId: env.AWS_ACCESS_KEY_ID,
+            secretAccessKey: env.AWS_SECRET_ACCESS_KEY,
+            sessionToken: env.AWS_SESSION_TOKEN,
+          },
+        }),
+        clearCredentialsAndIdentityId: () => {
+          /* noop */
+        },
+      },
+    },
+  }
+);
+
+const client = generateClient<Schema>({
+  authMode: "iam",
+});
+
+export const handler: PostConfirmationTriggerHandler = async (event) => {
+  await client.models.UserProfile.create({
+    email: event.request.userAttributes.email,
+  });
+
+  return event;
+};
+
+
+```
+The `amplifyconfiguration.json` file is not automatically accessible to your function. To include it with your function, you must first do a deployment of your backend using either `npx amplify sandbox` or Amplify hosting. Then copy the `amplifyconfiguration.json` file that is placed in your project root into your function directory. Now you can import the file into your function as shown above.
+
+<Callout warning>
+
+**Note:** Whenever you update your data model, you will need to copy the `amplifyconfiguration.json` file again.
+
+</Callout>
+
+Next, update the `amplify/data/resource.ts` file to define a data model for the user's profile:
+
+```ts title="amplify/data/resource.ts"
+import { type ClientSchema, a, defineData } from '@aws-amplify/backend';
+import { postConfirmation } from '../auth/post-confirmation/resource';
+
+
+const schema = a.schema({
+  UserProfile: a.model({
+    email: a.string(),
```

Comment:
I would go as far as saying this field in particular should also be locked down to only the function resource and the owner